DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/convolution2d/quartus_project_files/software/oneapi_2dconvolution.hpp

```diff
@@ -0,0 +1,153 @@
+
+#define KERNEL_STATUS_REGISTER              \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+  ID_CONVOLUTION2D_REGISTER_MAP_STATUS_REG)
+
+#define KERNEL_VERSION                         \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+  ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_VERSIONCSR_PIPE_CHANNEL_DATA_REG)
+
+#define KERNEL_VERSION_VALID                   \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+  ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_VERSIONCSR_PIPE_CHANNEL_VALID_REG)
+
+#define KERNEL_BYPASS                       \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_BYPASSCSR_PIPE_CHANNEL_DATA_REG)
+#define KERNEL_BYPASS_VALID                 \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ACL_C_ID_BYPASSCSR_PIPE_CHANNEL_VALID_REG)
+
+#define ARG_COEFFS                      \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ARG_ARG_COEFFS_0_REG)
+
+#define ARG_ROWS                            \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ARG_ARG_ROWS_REG)
+
+// add 4 to macro to account for a bug in the header file generated by SYCL HLS
+#define ARG_COLS                            \
+  (M_VVP_PIPELINE_ONEAPI_CONVOLUTION_BASE + \
+   ID_CONVOLUTION2D_REGISTER_MAP_ARG_ARG_COLS_REG + 4)
+
+
+namespace oneapiconvolution {
+
+// clang-format off
+#define K_COEFFS_SOBEL_VERTICAL        \
+  {                                    \
+    -1.0f / 6.0f, 0.0f, 1.0f / 6.0f,   \
+    -1.0f / 6.0f, 0.0f, 1.0f / 6.0f,   \
+    -1.0f / 6.0f, 0.0f, 1.0f / 6.0f    \
+  }
+
+#define K_COEFFS_IDENTITY   \
+  {                         \
+    0.0f, 0.0f, 0.0f,       \
+    0.0f, 1.0f, 0.0f,       \
+    0.0f, 0.0f, 0.0f        \
+  }
+
+void start() {
+  IOWR_32DIRECT(KERNEL_STOP, 0, 0x0);
+  IOWR_32DIRECT(KERNEL_STOP_VALID, 0, 0x1);
+  IOWR_32DIRECT(KERNEL_START, 0, 0x1);
+}
+
+void stop() {
+  IOWR_32DIRECT(KERNEL_STOP, 0, 0x1);
+  IOWR_32DIRECT(KERNEL_STOP_VALID, 0, 0x1);
+}
+
+int getStopCount() {
+  return IORD_32DIRECT(KERNEL_FINISH_COUNTER, 0);
+}
+
+// set convolution kernel
+
+void set_convolution_coeffs(size_t len, uint8_t *coeffs)
+{
+  printf("Setting coefficients:\n\t");
+  // copy byte-wise to register
+  for(size_t i = 0; i < len; i++) {
+    printf("0x%x ", coeffs[i]);
+    IOWR_8DIRECT(ARG_COEFFS + i, 0, coeffs[i]);
+  }
+  printf("\n");
+}
+
+void init_coeffs_identity3x3() {
+  float coeffs[9] = K_COEFFS_IDENTITY;
+
+  // re-cast to int to be compatible with the macro
+  uint8_t *coeffs_bytes = reinterpret_cast<uint8_t *>(coeffs);
+  set_convolution_coeffs(9 * sizeof(float), coeffs_bytes);
+}
+
+
+void init_coeffs_sobel_vertical() {
+  float coeffs[9] = K_COEFFS_SOBEL_VERTICAL;
+
+  // re-cast to int to be compatible with the macro
+  uint8_t *coeffs_bytes = reinterpret_cast<uint8_t *>(coeffs);
+  set_convolution_coeffs(9 * sizeof(float), coeffs_bytes);
+}
+
+uint64_t pollStatusRegister() {
```

Comment:
Is there a reason to switch to camelCase for this group of functions?

Reply:
These functions are embedded nios code that runs in a soft CPU, and follows a different coding standard.